users/utils.py

```diff
@@ -0,0 +1,36 @@
+"""User app utility functions"""
+import re
+
+from users.constants import USERNAME_MAX_LEN
+
+
+USERNAME_INVALID_CHAR_PATTERN = r"[^\w]|[\d]"
+
+
+def usernameify(full_name):
+    """
+    Generates a username based on a full name
+
+    Args:
+        full_name (str): A full name (i.e.: User.name)
+
+    Returns:
+        str: A generated username
+    """
+    return re.sub(USERNAME_INVALID_CHAR_PATTERN, "", full_name).lower()[
```

Comment:
I think maybe we should replace any whitespace with a single underscore so the usernames are more legible, plus probably a `trim` to ensure no trailing or leading spaces

Reply:
Definitely agree with `trim`. I can see some downsides w/ the underscores though. For one, we could end up having some usernames like `john_doe_12` and others like `john_doe12`, and "titles" (for lack of a better term) could end up being handled awkwardly, e.g.: "Sr" or "Esq" could yield a username that ends with something odd like "_s" or "_e". You'd probably call these edge cases since we're not likely to have many users with full names that exceed 26 characters (even with spaces). I'm not sure it would be that jarring for any user to see there name spelled out unbroken in lowercase, though. @pdpinch any thoughts on this?

Reply:
In that case, we should widen the column. According to https://www.postgresql.org/docs/9.2/release-9-2.html#AEN114949, such a migration won't lock the table, so we can safely do this. The 26 char limit was because that was the size of a `ulid`. Picking a semi-arbitrary number, 50 chars would probably be enough?